backend/src/bin/api.rs

```diff
@@ -46,9 +54,53 @@ async fn main() -> Result<(), Box<dyn Error>> {
 
 /// Create a SQLite database if needed, then connect to it and run migrations.
 /// Return a connection pool.
-async fn create_sqlite_pool() -> Result<SqlitePool, Box<dyn Error>> {
+async fn create_sqlite_pool(
+    reset_database: bool,
+    load_fixtures: bool,
+) -> Result<SqlitePool, Box<dyn Error>> {
     let opts = SqliteConnectOptions::from_str(DB_URL)?.create_if_missing(true);
+    if reset_database {
+        // remove the file, ignoring any errors that occured (such as the file not existing)
+        let _ = tokio::fs::remove_file(opts.get_filename()).await;
+    }
     let pool = SqlitePool::connect_with(opts).await?;
     sqlx::migrate!().run(&pool).await?;
+
+    if load_fixtures {
+        fixtures::seed_fixture_data(&pool).await?;
+    }
+
     Ok(pool)
 }
+
+mod fixtures {
+    macro_rules! load_fixture {
+        ($fixture:literal) => {
+            Fixture {
+                data: include_str!(concat!("../../fixtures/", $fixture, ".sql")),
```

Comment:
It would be nice to be able to give the path to the fixtures as an optional argument. That way when the pipeline uploads and then downloads the built executable and the fixtures, they don't have to be in a folder structure matching this path.

Reply:
[`include_str!`](https://doc.rust-lang.org/std/macro.include_str.html) includes the file in the binary, so it doesn't need to be present when running the binary. I think that should resolve this?

Reply:
Yep! That should just work, you don't need the fixture files anymore after you've produced the binary. I could sort of see the use-case of allowing you to load different sets of fixtures for different use-cases, but I would prefer to only add that if we ever need it.

Reply:
Ah cool, that makes it even easier. :-)

That does mean that for now we would ship the fixtures in a release?

Reply:
I can put it behind a feature flag that is enabled by default, that way we can disable all the database loading and resetting functionality for a build that is intended for release? @praseodym any opinion?

Reply:
> I can put it behind a feature flag that is enabled by default, that way we can disable all the database loading and resetting functionality for a build that is intended for release? @praseodym any opinion?

Sounds good!